proposals/0293-extend-property-wrappers-to-function-and-closure-parameters.md

```diff
@@ -16,193 +52,289 @@ Property Wrappers were [introduced in Swift 5.1](https://github.com/apple/swift-
-2. The parameter name will be prefixed with an underscore.
-3. The type of the parameter will be the backing property-wrapper type.
-4. A local computed property representing the `wrappedValue` of the innermost property wrapper will be synthesized with the same name as the original, unprefixed parameter name. If the innermost `wrappedValue` defines a setter, a setter will be synthesized for the local property if the mutability of the composed setter is `nonmutating`. The mutability computation is specified below.
-5. If the outermost property wrapper defines a `projectedValue` property, a local computed property representing the outermost `projectedValue` will be synthesized and named per the original parameter name prefixed with a dollar sign (`$`). If the outermost `projectedValue` defines a setter, a setter for the local computed property will be synthesized if the `projectedValue` setter is `nonmutating`, or if the outermost wrapper is a reference type.
+Property wrappers can opt into passing a projected-value argument to a property-wrapped parameter.
 
-#### Mutability of composed `wrappedValue` accessors
+Though property-wrapper projections can be utilized to expose arbitrary API through the synthesized `$` property, projections are typically used to either publicly expose the backing property wrapper directly, or to provide a public representation of the backing wrapper that's suitable for use outside of the declaration that owns the wrapper storage. In such cases, supporting property-wrapper initialization from a projected-value is very useful, especially if the wrapper does not support `init(wrappedValue:)`. To support passing a property-wrapper projection to a function with a wrapped parameter, property wrappers can implement `init(projectedValue:)`.
 
-The computation for mutability of a wrapped parameter's composed `wrappedValue` accessors will be the same as it is today for wrapped properties. The computation starts with the mutability of the outermost wrapper's `wrappedValue` accessor, and then iterates over the chain of composed property wrappers, "composing" the mutability of each `wrappedValue` accessor along the way using the following rules, which are the same for getters and setters:
+Presence of an `init(projectedValue:)` that meets the following requirements enables passing a projected value via the `$` calling syntax:
 
-* If the next `wrappedValue` accessor is `nonmutating`, then the mutability of the composed accessor is the same as the previous composed getter. If the wrapper is a reference type, the accessor is considered `nonmutating`.
-* If the next `wrappedValue` accessor is `mutating`, then the composed accessor is `mutating` if the previous composed getter _or_ setter is `mutating`, since both are needed to perform a writeback cycle.
+- The first parameter of this initializer must be labeled `projectedValue` and have the same type as the `var projectedValue` property.
```

Comment:
If I understand correctly, declaring a `projectedValue` property is not mandatory; should we perhaps state this explicitly? 

We should probably add a rationale for this behavior, as well.

Reply:
My intention was that `var projectedValue` is mandatory for `init(projectedValue:)` to enable the `$` calling syntax. This obviously is not clear, so yes, we should explicitly state this in the proposal!

Reply:
Fixed!